### PR TITLE
08 Place Order Back End | Yang An Boada

### PR DESCRIPTION
**Pull request by yanganboada**
wicked-08

### CODE REVIEW

File: server/index.js [hunk 1]
```diff
@@ -176,7 +176,39 @@ app.post('/api/cart', (req, res, next) => {
...
+        insert into "orders" ("name", "creditCard", "shippingAddress", "cartId")
+        values ($1, $2, $3, $4)
+        returning *;
+      `;
+
+      const params = [ nameVal, creditCardVal, shippingAddVal, cartIdVal];
 
+      db.query(insert, params)
+        .then(result => {
+          delete req.session.cartIdVal;
```
Comment (thebearingedge) [inline server/index.js]: This should be `cartId`.

File: server/index.js [hunk 2]
```diff
@@ -176,7 +176,39 @@ app.post('/api/cart', (req, res, next) => {
...
+  const creditCardVal = req.body.creditCard;
+  const shippingAddVal = req.body.shippingAddress;
+
+  if (!cartIdVal) {
+    throw new ClientError('Invalid "cartId" or "cartId is not existed', 400);
+  } else {
+    if (!nameVal || !creditCardVal || !shippingAddVal) {
+      throw new ClientError('"name", "creditCard" and "shippingAddress" are all required', 400);
+    } else if (!Number(creditCardVal) || `${creditCardVal}`.length !== 15 && `${creditCardVal}`.length!==16) {
+      throw new ClientError('Credit Card number must be 15 or 16 digits', 400);
```
Comment (thebearingedge) [inline server/index.js]: Credit card numbers are actually not numbers, they are strings.

**Review by thebearingedge — CHANGES_REQUESTED** [review]

File: server/index.js (hunk 1 above)
Comment (yanganboada) [reply server/index.js]: Updated 
b6f87a7

File: server/index.js (hunk 2 above)
Comment (yanganboada) [reply server/index.js]: Updated 
b6f87a7

File: server/index.js [hunk 3]
```diff
@@ -176,7 +176,39 @@ app.post('/api/cart', (req, res, next) => {
...
+        insert into "orders" ("name", "creditCard", "shippingAddress", "cartId")
+        values ($1, $2, $3, $4)
+        returning *;
+      `;
+
+      const params = [ nameVal, creditCardVal, shippingAddVal, cartIdVal];
 
+      db.query(insert, params)
+        .then(result => {
+          delete cartIdVal;
```
Comment (thebearingedge) [inline server/index.js]: This needs to be deleted from the `req.session` (`cartId`). Deleting the variable does not affect the `req.session` object.

**Review by thebearingedge — CHANGES_REQUESTED** [review]

File: server/index.js (hunk 3 above)
Comment (yanganboada) [reply server/index.js]: I see, updated with 
00e2d78

**Review by thebearingedge — APPROVED** [review]

**Pull request merged**